src/parser/Debug_Options.cc

```diff
@@ -102,10 +144,87 @@ string debug_options_as_text(unsigned const debug_options) {
   if (debug_options & DEBUG_RESET_TIMING) {
     Result += ", RESET_TIMING";
   }
+  // Mask out standard options and see if any extensions are active
+  debug_options = debug_options &
+                  ~(DEBUG_ALGORITHM | DEBUG_TIMESTEP | DEBUG_TIMING |
+                    DEBUG_RESET_TIMING | DEBUG_BALANCE | DEBUG_MEMORY);
+
+  if (debug_options) {
+    for (auto i = extended_debug_option.begin();
+         i != extended_debug_option.end(); ++i) {
+      if (debug_options & i->second) {
+        Result += ", " + i->first;
+      }
+    }
+  }
 
   return Result;
 }
 
+//---------------------------------------------------------------------------------------//
+/*! Add a new debug option to the debug parser specific to an application. This version
+    assigns the next available bit.
+ *
+ * \param Debug option keyword
```

Comment:
Can you use `\param[in] option_name Debug option keyword`? 
* Doxygen expects the variable name to be listed. 
* The [in] out looks nice in the docs. 
* Ditto for remaining function documentation below.